integration-tests/contract-security-wrong-field-annotations.spec.ts

```diff
@@ -0,0 +1,275 @@
+ * callWrongAnnot       (pair %callWrongAnnot (string %entrypoint) (pair %param (nat %a) (nat %b)))
+ * 
+ * The test checks 9 cases: For each call entrypoint the tests call each of the three add entrypoints.
+ * Its not entriely clear what should happen in each case. 
+ * 
+ * For example, 
+ *    the first case calls "add" and passes nat %valueA and nat %valueB to CONTRACT %add (pair (nat %valueA) (nat %valueB)), so the annots and params match 
+ *    the second case calls "addNoAnnot" and passes nat and nat to CONTRACT %addNoAnnot (pair (nat %valueA) (nat %valueB)), and here the  annots and params 
+ *    do not match because there are no annots.
+ *    the third case calls "addWrongAnnot" and passes nat a and nat b (where a and b are the annots) to CONTRACT %add (pair nat nat) and here the annots 
+ *    and params do not match.
+ * 
+ *    When run the tests get unexpected exceptions such as 
+ *        [a] Value is not a number: undefined
+ *        [valueA] Value is not a number: undefined
+ * 
+ *  With Jakarta there is a change regarding Annotations: https://tezos.gitlab.io/protocols/013_jakarta.html#michelson. 
+ *  This means the sub testcases 3 & 7 in TC-009 will behave differently in Jakarta.
+ *        
+ */
+
+CONFIGS().forEach(({ lib, rpc, setup, protocol }) => {
+  const Tezos = lib;
+  const mondaynet = protocol === Protocols.ProtoALpha ? test: test.skip;
+
+  describe(`Test contracts to verify wrong field annotations are leading to failed transactions using: ${rpc}`, () => {
+    beforeEach(async (done) => {
+      await setup();
+      done();
+    });
+
+    mondaynet('Verify annotation combinations on ithacanet', async () => {
+      const addition = await Tezos.contract.originate({
+        code: securityWrongAnnotations,
+        init: `0`,
+      });
+      await addition.confirmation();
+      const contractAddition = await addition.contract();
+
+      //Testcase #1
+      //Calls entrypoint: (pair %add (nat %valueA) (nat %valueB))
+      //Calling Michelson code: CONTRACT %add (pair (nat %valueA) (nat %valueB)) ;
+
+      const op1 = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .call({
+              entrypoint: 'add',
+              param: {
+                valueA: `1`,
+                valueB: `2`,
+              },
+            })
+            .send();
+        })
+        .then((op1) => {
+          return op1.confirmation().then(() => op1.hash);
+        });
+
+      //Testcase #2
+      // Calls entrypoint: (pair %addNoAnnot nat nat))
+      // Calling Michelson code: CONTRACT %addNoAnnot (pair (nat %valueA) (nat %valueB)) ;
+      try {
+        const op2 = await Tezos.contract
+          .at(contractAddition.address)
+          .then((contract) => {
+            return contract.methodsObject
+              .call({
+                entrypoint: 'addNoAnnot',
+                param: {
+                  0: `1`,
```

Comment:
Taquito returns error `'[valueA] Value is not a number: undefined'` because the annotations is incorrect.
Instead of index 0 , it should be valueA
Instead of index 1, it should be valueB

Reply:
I made the change but it still gives the same error,.